### PR TITLE
Add random character selection, controls help option and fix keyboard mapping

### PR DESCRIPTION
**Pull request by vitorbaraujo**
Added features:
- Random character selection
- Indicator to start battle
- Controls help option
- Fix keyboard mapping

### CODE REVIEW

File: src/BattleState.cpp [hunk 1]
```diff
@@ -67,27 +66,21 @@ BattleState::BattleState(string stage, string cmusic, vector< pair<string, strin
...
+		input_manager->key_press(InputManager::K_SELECT) ||
+		input_manager->joystick_button_press(InputManager::SELECT, 0)
+	){
 		music.stop();
 		m_quit_requested = true;
 		Game::get_instance().push(new MenuState());
 		return;
 	}
 
-	if(input_manager->quit_requested()){
```
Comment (joao18araujo) [inline src/BattleState.cpp]: This if should not be removed, it closes the window when pressed Alt+F4 or when clicked the close window button. The other way, it returns to the menu, which is wrong

File: src/EditState.cpp [hunk 2]
```diff
@@ -33,42 +32,51 @@ EditState::EditState(string cstage) : stage(cstage){
...
+	if(input_manager->quit_requested() ||
+		input_manager->key_press(InputManager::K_SELECT) ||
+		input_manager->joystick_button_press(InputManager::SELECT, 0)
+	){
 		m_quit_requested = true;
 		Game::get_instance().push(new MenuState());
 		return;
 	}
 
-	if(input_manager->quit_requested()){
```
Comment (joao18araujo) [inline src/EditState.cpp]: If should not be removed too

**Review by joao18araujo — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -8,4 +8,5 @@ install:
 compiler:
    - g++
 script:
-  - make
+  - make dist-clean
```
Comment (joao18araujo) [inline .travis.yml]: Travis file should not be changed

**Review by joao18araujo — CHANGES_REQUESTED** [review]

File: src/BattleState.cpp [hunk 4]
```diff
@@ -67,27 +66,21 @@ BattleState::BattleState(string stage, string cmusic, vector< pair<string, strin
...
-//		delete(fighters[i]);
-
-//	fighters.clear();
 }
 
 void BattleState::update(float delta){
 	InputManager * input_manager = InputManager::get_instance();
 
-	if(input_manager->key_press(SDLK_ESCAPE) or input_manager->joystick_button_press(InputManager::SELECT, 0)){
+	if(input_manager->quit_requested() ||
```
Comment (joao18araujo) [inline src/BattleState.cpp]: input_manager->quit_requested() should be on another if

**Review by joao18araujo — CHANGES_REQUESTED** [review]

File: src/BattleState.cpp (hunk 1 above)
Comment (vitorbaraujo) [reply src/BattleState.cpp]: Done.

File: src/BattleState.cpp (hunk 4 above)
Comment (vitorbaraujo) [reply src/BattleState.cpp]: Done.

File: .travis.yml (hunk 3 above)
Comment (vitorbaraujo) [reply .travis.yml]: Done.

File: src/EditState.cpp (hunk 2 above)
Comment (vitorbaraujo) [reply src/EditState.cpp]: Done.

**Review by joao18araujo — APPROVED** [review]

**Pull request merged**